packages/land/contracts/LandV4.sol

```diff
@@ -0,0 +1,196 @@
+// SPDX-License-Identifier: MIT
+/* solhint-disable no-empty-blocks */
+pragma solidity 0.8.23;
+
+import {LandBaseTokenV4} from "./mainnet/LandBaseTokenV4.sol";
+import {ERC721BaseTokenV3} from "./mainnet/ERC721BaseTokenV3.sol";
+import {OperatorFiltererUpgradeable} from "./mainnet/OperatorFiltererUpgradeable.sol";
+import {RoyaltyDistributorV2} from "@sandbox-smart-contracts/dependency-royalty-management/contracts/RoyaltyDistributorV2.sol";
+import {IOperatorFilterRegistry} from "./mainnet/IOperatorFilterRegistry.sol";
+import {IERC2981Upgradeable} from "@openzeppelin/contracts-upgradeable/interfaces/IERC2981Upgradeable.sol";
+
+/**
+ * @title LandV4
+ * @author The Sandbox
+ * @notice LAND contract
+ * @dev LAND contract implements ERC721, quad and marketplace filtering functionalities
+ */
+contract LandV4 is LandBaseTokenV4, OperatorFiltererUpgradeable, RoyaltyDistributorV2 {
+    bool internal _initV4;
+
+    event OperatorRegistrySet(address indexed registry);
+
+    modifier initializerV4() {
+        require(!_initV4, "already initialized");
+        _;
+    }
+
+    /**
+     * @notice Initializes the upgrade to version 4
+     * @param _royaltyManager address of the manager contract for common royalty recipient
+     */
+    function initializeV4(
+        address metaTransactionContract,
+        address admin,
+        address _royaltyManager
+    ) public initializerV4 {
```

Comment:
Can we try to use a generic version with `reinitializer` ? So that the deploy package is responsible of the versioning

```
contract MyToken is MyToken, ERC20PermitUpgradeable {
    function initialize(uint256 version) reinitializer(version) public {
        __ERC20Permit_init("MyToken");
    }
}
```